drivers/network/afd/afd/lock.c

```diff
@@ -328,17 +330,55 @@ PAFD_HANDLE LockHandles( PAFD_HANDLE HandleArray, UINT HandleCount ) {
             continue;
         }
 
+        // Get the object pointer, as UserMode.
         Status = ObReferenceObjectByHandle((HANDLE)HandleArray[i].Handle,
+                                           0,
+                                           NULL,
+                                           UserMode,
+                                           &Object,
+                                           NULL);
+        if (!NT_SUCCESS(Status))
+        {
+            AFD_DbgPrint(MIN_TRACE,
+                         ("ObReferenceObjectByHandle(%p, UserMode) failed: 0x%08lx\n",
+                          (HANDLE)HandleArray[i].Handle, Status));
+
+            UnlockHandles(FileObjects, i);
+            return NULL;
+        }
+
+        // Create a kernel handle from the pointer.
+        Status = ObOpenObjectByPointer(Object,
```

Comment:
Your logic here is
* reference handle U to get object O
* create handle K from object O
* reference handle K to get object O

That's entirely pointless, just save the object the first time.

Reply:
> pointless

Existing code uses 'KernelMode, FILE_ALL_ACCESS'...
Now replaced with 'UserMode, 0'.